File: python/questions/evalRPN.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for token in tokens:
            if token == "+" or token == "-" or token == "*" or token == "/":
                num = st.pop()
                if token == "+":
                    st[-1] += num
                elif token == "-":
                    st[-1] -= num
                elif token == "*":
                    st[-1] *= num
                elif token == "/":
                    st[-1] = trunc(st[-1]/num)
                else:
                    st[-1] = st[-1]
            else:
                st.append(int(token))
        return st[-1]

File: cpp/questions/evalRPN.cpp
class Solution {
public:
    int evalRPN(vector<std::string>& tokens) {
        std::stack<int> st;
        char t0;
        int num;
        for (std::string& token : tokens) {
            t0 = token[0];
            if (token.size() > 1 || std::isdigit(t0)) {
                st.push(std::stoi(token));
                continue;
            }
            num = st.top();
            st.pop();
            if (t0 == '+') {
                st.top() += num;
            } else if (t0 == '-') {
                st.top() -= num;
            } else if (t0 == '*') {
                st.top() *= num;
            } else if (t0 == '/') {
                st.top() /= num;
            }
        }
        return st.top();
    }
};
